### PR TITLE
Scibert reranking

### PR DESCRIPTION
**Pull request by aCampello**

### CODE REVIEW

File: covid/question_covid.py [hunk 1]
```diff
@@ -51,20 +60,68 @@ def get_answer(self, text, question):
...
+                        for i, tf_idf_score, text in list_of_texts]
+        similarities = sorted(similarities, key=lambda x: x[2], reverse=True)
+
+        return similarities
+
+    def _scibert_embedding(self, x):
+        # Max sequence length is 512 for BERT
+        if len(x) > 512:
+            embedded_a = self._scibert_embedding(x[:512])
+            embedded_b = self._scibert_embedding(x[512:])
```
Comment (lizgzil) [inline covid/question_covid.py:83]: so the second part of the text is very likely to be <512 in length?

File: covid/question_covid.py [hunk 2]
```diff
@@ -51,20 +60,68 @@ def get_answer(self, text, question):
...
+        # Re-ranks everything from Scibert
+        best_match_texts = self.rank_shortlisted(list_of_texts=best_match_texts,
+                                                 question=question)
+
+        for i, tfidf_score, scibert_score, text in best_match_texts[:n_scibert_matches]:
+            best_score = 0  # if score is negative, i consider the answer wrong
+            best_answer = "No answer"
+            best_text = "No snippet"
+
+            n = 3
```
Comment (lizgzil) [inline covid/question_covid.py:123]: unneeded now

**Review by lizgzil — APPROVED** [review]
looks good

File: covid/question_covid.py (hunk 1 above)
Comment (aCampello) [reply covid/question_covid.py:83]: No, if the second part is bigger again, it will call the function recursively again

Comment (aCampello) [reply covid/question_covid.py:83]: @nsorros's masterpiece 🤓

Comment (nsorros) [reply covid/question_covid.py:83]: far from masterpiece, haha

Comment (lizgzil) [reply covid/question_covid.py:83]: oo that's cool. function inception

**Pull request merged**